dxp/javascript_minification.md

```diff
@@ -0,0 +1,86 @@
+
+### YahooJavaScriptMinifier
+
+This uses [YUI Compressor](https://github.com/yui/yuicompressor) from Yahoo!, which is still active but unmaintained (except for some PRs that get merged from time to time). The last commit at the time of this writing (Sep 2020) is from May 2019.
+
+Unless someone has a very good reason to use it, this shouldn't be the choice for any production site but it is still shipped for compatibility reasons.
+
+## Components of DXP using the minifier
+
+Currently, the following components in DXP use the minifier services.
+
+### StripFilter
+
+This is a filter used to minify inline JavaScript code contained in `<script>` tags and embedded in the HTML code.
+
+> It also does transformations for `<input>`, `<textarea>`, `<pre>` and `<style>` but that's out of the scope of this document.
+
+It filters any resource with its URL starting with (as specified in [liferay-web.xml](https://github.com/liferay/liferay-portal/blob/master/portal-web/docroot/WEB-INF/liferay-web.xml)):
+
+-   `/c/...`
+-   `/group/...`
+-   `/user/...`
+-   `/web/...`
+
+And it can be configured with:
+
+-   [portal.properties](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/portal.properties):
+    -   `minifier.inline.content.cache.enabled`: set to `true` to make the filter minify `<script>` tags content (by default `true`).
+    -   `minifier.inline.content.cache.skip.javascript`: this defines a comma separated list of tokens to look for in the `<script>` content so that, if any is found, the minified response is not cached. By default, any `<script>` containing `getSessionId` or `encryptedUserId` is not cached.
+    -   `strip.ignore.paths`: list of paths to ignore (by default `/document_library/get_file`).
+    -   `strip.mime.types` in `portal.properties`: defines the mime types of responses that should be processed by the filter (by default `text/html*` and `text/xml*`). Note that you can put a `*` at the end of the config values to catch all mime types starting with the string (not being exactly the string).
+-   Request parameters:
+    -   `ensureContentLength`: set to `true` if you want to make sure the `Content-Length` header is returned from the server.
+    -   `strip`: set to `false` to disable the filter.
+    -   `p_p_lifecycle`: if it is `1` ([PortletRequest.ACTION_PHASE](https://docs.oracle.com/cd/E13155_01/wlp/docs103//javadoc/javax/portlet/PortletRequest.html#ACTION_PHASE)) or `2` ([PortletRequest.RESOURCE_PHASE](https://docs.oracle.com/cd/E13155_01/wlp/docs103//javadoc/javax/portlet/PortletRequest.html#RESOURCE_PHASE)) the filter is disabled.
+
+If all the rules match and a `<script>` is minified, the filter simply invokes [MinifierUtil.minifyJavaScript()](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/minifier/MinifierUtil.java#L45) to do its duty.
```

Comment:
Exact line number here suggest you need "y" to make the link robust against future changes in `master`.